lib/main_containers.pm

```diff
@@ -31,6 +31,11 @@ sub is_container_test {
     get_var('CONTAINER_RUNTIME') ? return 1 : return 0;
 }
 
+sub is_container_image_test {
+    my $flavor = get_required_var('FLAVOR');
+    return grep { /$flavor/ } ('Container-Image', 'Container-Image-Updates', 'NET');
```

Comment:
Doesn't this work too?
$flavor =~ /(Container-Image|Container-Image-Updates|NET)/

Reply:
yes

Reply:
If I am not wrong, this way it would match the initial 3 cases `/(Container-Image(-Updates)?|NET)/` as we are using string literals.

Reply:
`/(Container-Image|NET)/` matches  `Container-Image` and whatever you append, e.g. `Container-ImageXYZ`. So `Container-Image-Updates` is matched.

Reply:
> `/(Container-Image|NET)/` matches `Container-Image` and whatever you append, e.g. `Container-ImageXYZ`. So `Container-Image-Updates` is matched.

yes, you are right! Thanks!

```
Container-Image|Container-Image-Updates|NET matched NET
Container-Image|Container-Image-Updates|NET matched Container-Image
Container-Image|Container-Image-Updates|NET matched Container-Image-Updates

Container-Image|NET matched NET
Container-Image|NET matched Container-Image
Container-Image|NET matched Container-Image-Updates
```